var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -172,6 +172,7 @@ class Boost(Package):
     patch('call_once_variadic.patch', when='@1.54.0:1.55.9999%gcc@5.0:5.9')
 
     # Patch fix for PGI compiler
+    patch('boost_pgi.patch', when='%pgi')
```

Comment:
Did you confirm that this patch applies to all versions of boost in Spack?

Reply:
I have not.  Trying to do so, I've found that some of the existing patches in Spack don't apply to all versions of Boost.  I will open an issue on this.

Reply:
Don't bother opening an issue, just add a `when` clause to those patches for the versions they do work for. You can do this in the same PR.

Reply:
Okay, will do.

Reply:
To be honest @adamjstewart, those PGI changes shouldn't even be in this patch.  I'm going to pull them out.